### PR TITLE
simplify Outbuffer.vprintf

### PR DESCRIPTION
**Pull request by rainers**
see https://github.com/dlang/dmd/pull/10525

### CODE REVIEW

File: src/dmd/root/outbuffer.d [hunk 1]
```diff
@@ -299,42 +299,25 @@ struct OutBuffer
...
+            va_list va;
+            va_copy(va, args);
+            /*
+                The functions vprintf(), vfprintf(), vsprintf(), vsnprintf()
+                are equivalent to the functions printf(), fprintf(), sprintf(),
+                snprintf(), respectively, except that they are called with a
+                va_list instead of a variable number of arguments. These
+                functions do not call the va_end macro. Consequently, the value
+                of ap is undefined after the call. The application should call
+                va_end(ap) itself afterwards.
```
Comment (rainers) [inline src/dmd/root/outbuffer.d]: BTW: I think this comment only applies if `vsnprintf` is a macro in C and changes the `va` argument, but that is not possible in the D version as it is always a function call. I think the copying of `args` could be removed, too.

Comment (WalterBright) [reply src/dmd/root/outbuffer.d]: I don't recall why that comment was written. The va_copy() is necessary to initialize va.

**Review by WalterBright — APPROVED** [review]

File: src/dmd/root/outbuffer.d (hunk 1 above)
Comment (rainers) [reply src/dmd/root/outbuffer.d]: The comment is more or less a copy from the specification at cppreference.com. `args` already is a `va_list`, no need to copy it unless it is modified by every iteration of the for-loop. That doesn't happen as a copy is passed to vsnprintf anyway.

Comment (kinke) [reply src/dmd/root/outbuffer.d]: I think it's no copy for Posix x86_64, where the va_list type has reference semantics IIRC.

Comment (rainers) [reply src/dmd/root/outbuffer.d]: Indeed, `va_copy` creates and fills some stack allocated additional data, there, so better left as is.

**Review by kinke — APPROVED** [review]

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**